### ISSUE TITLE
select_level default

### ISSUE THREAD

**Issue by hbrunn** [OCA/openupgradelib#123]
I'm using `add_fields` on a v8 database, but this breaks because select_level is [required](https://github.com/OCA/OCB/blob/8.0/openerp/addons/base/ir/ir_model.py#L261) there and there's [no default](https://github.com/OCA/OCB/blob/8.0/openerp/addons/base/base.sql#L31). This field doesn't exist in newer versions, so we can't just change the [statement](https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L1899).

We could do some version detection and adapt the statement accordingly, but why don't we use the ORM here? The function wants an environment anyways, and if you've an environment, then `ir.model.fields` et al exist already in the registry.

TLDR: Why don't we use the ORM in `add_fields`?

**Comment by StefanRijnhart** [comment]
Well, add_fields is used in the 11.0 base module pre-migration script. Would adding the field in the ORM work there?

If I understand correctly, add_fields was developed in favour of just creating the SQL column to cover up for the fact that Odoo does not add an XML entry for fields with existing columns, which was fixed in Odoo 12.0. We might as well just backport that fix to OpenUpgrade 9, 10 and 11 (I got this link from the add_fields docstring: https://github.com/odoo/odoo/blob/9201f92a4f29a53a014b462469f27b32dca8fc5a/odoo/addons/base/models/ir_model.py#L794-L802)

**Comment by hbrunn** [comment]
thanks, base's pre-migration won't fly there.
Hmm, I'm not sure how sensible it is to put effort into making this work for v8. For the time being, I simply add a default before my migration as in

```
alter table ir_model_fields alter column select_level set default '0'
```

Should we just close the issue and keep this fix here for people who google the issue?

**Comment by pedrobaeza** [comment]
@hbrunn I added that method and I didn't checked on v8 certainly. On the docstring I should put >v8, not >=v8. It's done through SQL due to several reasons:

- For being able to use it on `base` module.
- For avoiding problems with ORM code attached to `ir.model.fields` model that tries to do some incompatible stuff, like forcing `x_` as preffix or so.

So the chances are those you are commenting:

- Adapt the method for adding that extra field on v8.
- Mention the workaround on docstring and let it as is. Less elegant...

**Comment by pedrobaeza** [comment]
I have changed the docstring as we don't have the other solution: 5a0a2369e08a141a9a5bc8f091c4fc0522b07fa3
